File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int>freq;
        int left=0;
        int ans=0;
        for(int right=0;right<s.length();right++){
            int ch=s[right];
            freq[ch]++;
            while(freq[ch]>1){
                freq[s[left]]--;
                left++;
            }
            ans=max(ans,right-left+1);
        }
        return ans;   
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
from collections import Counter
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mp=Counter()
        left=0
        right=0
        ans=0
        while right< len(s):
            mp[s[right]]+=1
            while(mp[s[right]]>1):
                mp[s[left]]-=1
                left+=1
            ans=max(ans,right-left+1)
            right+=1
        return ans